src/index.ts

```diff
@@ -133,3 +133,53 @@ export const ReadonlyDate = (
   // tslint:disable-next-line: max-union-size
   value: number | string | Date | ReadonlyDate
 ): ReadonlyDate => new Date(value as number | string | Date);
+
+export interface ReadonlyWeakKeyedCollection<K> {
+  /** Returns a boolean indicating whether an element with the specified value exists in the collection  */
+  readonly has: (key: K) => boolean;
+}
+
+export interface ReadonlyKeyedCollection<K, V>
+  extends ReadonlyWeakKeyedCollection<K> {
+  /** Calls callbackFn once for each key-value pair present in the collection, in insertion order. If a thisArg parameter is provided to forEach, it will be used as the this value for each callback. */
+  readonly forEach: (
+    callbackfn: (value: K, value2: V, set: this) => void,
```

Comment:
Using interfaces lets us use the `this` type which will be the correct specific type (e.g. ReadonlySet)